src/main/java/ua/softserveinc/tc/dao/BaseDao.java

```diff
@@ -1,17 +1,14 @@
 package ua.softserveinc.tc.dao;
 
-import javax.persistence.EntityManager;
 import java.util.List;
 
 public interface BaseDao<T> {
-    EntityManager getEntityManager();
+    T findById(Object id);
```

Comment:
I think we should consider to use ```Optional``` as a return type in methods that can return ```null```. This is a perfect use case for such situations.